File: Vitality.Website.App/Vacancies/Models/Vacancy.cs
using System;
using RestSharp.Deserializers;

namespace Vitality.Website.App.Vacancies.Models
{
    [DeserializeAs(Name = "item")]
    public class Vacancy
    {
        public int Advertid { get; set; }
        public string ClosingDate { get; set; }
        public string Description { get; set; }
        public string EmploymentType { get; set; }
        public string Joblocation { get; set; }
        public string Jobsalary { get; set; }
        public string Jobtitle { get; set; }
        public string Location { get; set; }
        public string Salary { get; set; }
        public string Title { get; set; }
    }
}

File: Vitality.Website/src/ts/app/Models/vacancy.ts
import { ExtraDataItem } from '../models/extradataitem';

export class Vacancy {
	Advertid: number;
	ClosingDate: string;
	Description: string;
	JobType: string;
	JobLocation: string;
	JobSalary: string;
    JobTitle: string;
	Link: string;
    Title: string;
    ExtraDataItems: Array<ExtraDataItem>;
}